monitored-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/HttpClientPool.java

```diff
@@ -88,44 +88,50 @@ private static void refreshTheList(
+                    .getList();
+
+            final String hostname = serverConfiguration.getHostname();
+
+            final Set<InetAddress> updatedLookup = dnsLookupWrapper.getInetAddressesByDnsLookUp(hostname);
+            if (updatedLookup.isEmpty()) {
+                if (oldListOfClients.isEmpty()) {
+                    LOGGER.log(Level.SEVERE, "The DNS lookup has returned an empty list of IPs. There is no client in the pool.");
+                } else {
+                    LOGGER.log(Level.WARNING, "The DNS lookup has returned an empty list of IPs. Reusing the old list.");
+                }
+                return;
             }
-            return;
-        }
-
 
-        httpClientsCache.set(new RoundRobinPool(
-                updatedLookup
-                        .stream()
-                        .map(inetAddress -> useOldClientOrCreateNew(
-                                singleHttpClientProvider,
-                                inetAddress,
-                                oldListOfClients,
-                                serverConfiguration,
-                                scheduledExecutorService
-                        ))
-                        .collect(Collectors.toUnmodifiableList())
-        ));
-
-        // Close those clients whose inet address is not present anymore
-        oldListOfClients.stream()
-                .filter(not(client -> updatedLookup.contains(client.getInetAddress())))
-                .forEach(oldClient -> {
-                    LOGGER.log(Level.INFO, () -> "The IP " + oldClient.getInetAddress().getHostAddress() + " for hostname " + hostname + " is not present in the DNS resolution list any more, closing the HttpClient");
-                    oldClient.close();
-                });
+            httpClientsCache.set(new RoundRobinPool(
+                    updatedLookup
+                            .stream()
+                            .map(inetAddress -> useOldClientOrCreateNew(
+                                    singleHttpClientProvider,
+                                    inetAddress,
+                                    oldListOfClients,
+                                    serverConfiguration,
+                                    scheduledExecutorService
+                            ))
+                            .collect(Collectors.toUnmodifiableList())
+            ));
+
+            // Close those clients whose inet address is not present anymore
+            oldListOfClients.stream()
+                    .filter(not(client -> updatedLookup.contains(client.getInetAddress())))
+                    .forEach(oldClient -> {
+                        LOGGER.log(Level.INFO, () -> "The IP " + oldClient.getInetAddress().getHostAddress() + " for hostname " + hostname + " is not present in the DNS resolution list any more, closing the HttpClient");
+                        oldClient.close();
+                    });
+        } catch (IllegalArgumentException e) {
```

Comment:
extracting it insied another method would ease the readability of the MR but also of the code in my opinion

Reply:
Extracted